# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/amdgpu-vm_update-cpu.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

[ 1789.485288] amdgpu 0000:04:00.0: amdgpu: Dumping IP State
[ 1789.486317] amdgpu 0000:04:00.0: amdgpu: Dumping IP State Completed
[ 1789.486427] amdgpu 0000:04:00.0: amdgpu: ring sdma0 timeout, signaled seq=15857, emitted seq=15859
[ 1789.486436] amdgpu 0000:04:00.0: amdgpu: Starting sdma0 ring reset
[ 1789.709846] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1789.709859] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1789.709867] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1789.709875] amdgpu 0000:04:00.0: amdgpu: GCVM_L2_PROTECTION_FAULT_STATUS:0x00101430
[ 1789.709880] amdgpu 0000:04:00.0: amdgpu:      Faulty UTCL2 client ID: SQC (data) (0xa)
[ 1789.709885] amdgpu 0000:04:00.0: amdgpu:      MORE_FAULTS: 0x0
[ 1789.709889] amdgpu 0000:04:00.0: amdgpu:      WALKER_ERROR: 0x0
[ 1789.709893] amdgpu 0000:04:00.0: amdgpu:      PERMISSION_FAULTS: 0x3
[ 1789.709898] amdgpu 0000:04:00.0: amdgpu:      MAPPING_ERROR: 0x0
[ 1789.709902] amdgpu 0000:04:00.0: amdgpu:      RW: 0x0
[ 1799.725290] amdgpu 0000:04:00.0: amdgpu: Dumping IP State
[ 1799.726202] amdgpu 0000:04:00.0: amdgpu: Dumping IP State Completed
[ 1799.726363] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1799.726374] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1799.726381] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1799.726389] amdgpu 0000:04:00.0: amdgpu: GCVM_L2_PROTECTION_FAULT_STATUS:0x00101431
[ 1799.726395] amdgpu 0000:04:00.0: amdgpu:      Faulty UTCL2 client ID: SQC (data) (0xa)
[ 1799.726401] amdgpu 0000:04:00.0: amdgpu:      MORE_FAULTS: 0x1
[ 1799.726406] amdgpu 0000:04:00.0: amdgpu:      WALKER_ERROR: 0x0
[ 1799.726411] amdgpu 0000:04:00.0: amdgpu:      PERMISSION_FAULTS: 0x3
[ 1799.726416] amdgpu 0000:04:00.0: amdgpu:      MAPPING_ERROR: 0x0
[ 1799.726421] amdgpu 0000:04:00.0: amdgpu:      RW: 0x0
[ 1799.726429] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1799.726436] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1799.726442] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1799.726480] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1799.726486] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1799.726492] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1799.726500] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1799.726506] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1799.726512] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1799.726558] amdgpu 0000:04:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:1 pasid:32771)
[ 1799.726564] amdgpu 0000:04:00.0: amdgpu:  in process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1799.726570] amdgpu 0000:04:00.0: amdgpu:   in page starting at address 0x0000800000c00000 from client 0x1b (UTCL2)
[ 1799.728024] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x00098560, 0x00015320, 0x00018580)
[ 1799.728266] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000A8540, 0x0001F9C0, 0x00028560)
[ 1799.728505] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000B8520, 0x0002F9A0, 0x00038540)
[ 1799.728752] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x00088500, 0x0003FD60, 0x00008520)
[ 1799.728999] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000984E0, 0x0000FD40, 0x00018500)
[ 1799.729238] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000A84C0, 0x0001F940, 0x000284E0)
[ 1799.729485] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000B84A0, 0x0002FD00, 0x000384C0)
[ 1799.729724] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x00088480, 0x0003F900, 0x000084A0)
[ 1799.729972] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x00098460, 0x0000FCC0, 0x00018480)
[ 1799.730212] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000A8440, 0x0001F8C0, 0x00028460)
[ 1799.730459] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x000B8420, 0x0002F8A0, 0x00038440)
[ 1799.730607] amdgpu 0000:04:00.0: amdgpu: ih ring buffer overflow (0x00082100, 0x0003CA00, 0x00002120)
[ 1799.732702] amdgpu 0000:04:00.0: amdgpu: ring gfx_0.1.0 timeout, but soft recovered
[ 1809.965292] amdgpu 0000:04:00.0: amdgpu: Dumping IP State
[ 1809.966063] amdgpu 0000:04:00.0: amdgpu: Dumping IP State Completed
[ 1809.976087] amdgpu 0000:04:00.0: amdgpu: ring gfx_0.1.0 timeout, signaled seq=252802, emitted seq=252805
[ 1809.976102] amdgpu 0000:04:00.0: amdgpu: Process information: process Hyprland pid 950 thread Hyprland:cs0 pid 956
[ 1809.976109] amdgpu 0000:04:00.0: amdgpu: Starting gfx_0.1.0 ring reset
[ 1810.197324] amdgpu 0000:04:00.0: amdgpu: Ring gfx_0.1.0 reset failure
[ 1810.197335] amdgpu 0000:04:00.0: amdgpu: GPU reset begin!
[ 1810.590675] amdgpu 0000:04:00.0: amdgpu: MODE2 reset
[ 1810.600850] amdgpu 0000:04:00.0: amdgpu: GPU reset succeeded, trying to resume
[ 1810.601451] [drm] PCIE GART of 1024M enabled (table at 0x000000F43FC00000).
[ 1810.601492] amdgpu 0000:04:00.0: amdgpu: PSP is resuming...
[ 1810.623490] amdgpu 0000:04:00.0: amdgpu: reserve 0xa00000 from 0xf43e000000 for PSP TMR
[ 1811.443497] amdgpu 0000:04:00.0: amdgpu: SMU is resuming...
[ 1811.444359] amdgpu 0000:04:00.0: amdgpu: SMU is resumed successfully!
[ 1811.445076] [drm] kiq ring mec 2 pipe 1 q 0
[ 1811.457437] [drm] DMUB hardware initialized: version=0x0300000A
[ 1811.534738] [drm] Failed to add display topology, DTM TA is not initialized.
[ 1811.919065] [drm] Failed to add display topology, DTM TA is not initialized.
[ 1812.191072] [drm] Failed to add display topology, DTM TA is not initialized.
[ 1812.236776] amdgpu 0000:04:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
[ 1812.236788] amdgpu 0000:04:00.0: amdgpu: ring gfx_0.1.0 uses VM inv eng 1 on hub 0
[ 1812.236795] amdgpu 0000:04:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 4 on hub 0
[ 1812.236800] amdgpu 0000:04:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 5 on hub 0
[ 1812.236804] amdgpu 0000:04:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
[ 1812.236808] amdgpu 0000:04:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
[ 1812.236812] amdgpu 0000:04:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
[ 1812.236816] amdgpu 0000:04:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
[ 1812.236820] amdgpu 0000:04:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
[ 1812.236825] amdgpu 0000:04:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
[ 1812.236829] amdgpu 0000:04:00.0: amdgpu: ring kiq_0.2.1.0 uses VM inv eng 12 on hub 0
[ 1812.236833] amdgpu 0000:04:00.0: amdgpu: ring sdma0 uses VM inv eng 13 on hub 0
[ 1812.236837] amdgpu 0000:04:00.0: amdgpu: ring vcn_dec_0 uses VM inv eng 0 on hub 8
[ 1812.236842] amdgpu 0000:04:00.0: amdgpu: ring vcn_enc_0.0 uses VM inv eng 1 on hub 8
[ 1812.236846] amdgpu 0000:04:00.0: amdgpu: ring vcn_enc_0.1 uses VM inv eng 4 on hub 8
[ 1812.236850] amdgpu 0000:04:00.0: amdgpu: ring jpeg_dec uses VM inv eng 5 on hub 8
[ 1812.241524] amdgpu 0000:04:00.0: amdgpu: GPU reset(4) succeeded!

From beec5cbd7260be325d8b20cc5314dfbd38332aae Mon Sep 17 00:00:00 2001
From: Friedrich Vock <friedrich.vock@gmx.de>
Date: Thu, 20 Jun 2024 14:12:15 +0200
Subject: [PATCH] drm/amdgpu: Don't use doorbells for SDMA 5.2

There seems to be a race condition and they can cause hangs in rare
scenarios.

Signed-off-by: Friedrich Vock <friedrich.vock@gmx.de>
(cherry picked from commit 9264ec783c738b272d94a72542524223b65255ef)
---
 drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c b/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
index c441a20a41a3..aa51e1c1b9e0 100644
--- a/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
+++ b/drivers/gpu/drm/amd/amdgpu/sdma_v5_2.c
@@ -1239,7 +1239,7 @@ static int sdma_v5_2_sw_init(void *handle)
 	for (i = 0; i < adev->sdma.num_instances; i++) {
 		ring = &adev->sdma.instance[i].ring;
 		ring->ring_obj = NULL;
-		ring->use_doorbell = true;
+		ring->use_doorbell = false;
 		ring->me = i;
 
 		DRM_INFO("use_doorbell being set to: [%s]\n",
-- 
2.47.0
